### PR TITLE
TST: Using  more fixtures in of tests/base/test_ops.py

### PR DESCRIPTION
**Pull request by SaturnFromTitan**
part of #23877
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

**Comment by jbrockmendel** [comment]
the diff looks a little mangled; is this all basically un-indent?

**Comment by SaturnFromTitan** [comment]
@jbrockmendel Not entirely. I had to extend the tests or add some extra conditions because the fixtures are more versatile than the original test values

**Comment by WillAyd** [comment]
Can these just be split into separate tests for index / series? One-offs are fine but this seems to branch a lot within the test based on the type of object, which makes the purpose of the test less clear

**Comment by SaturnFromTitan** [comment]
@WillAyd I guess you're referring to `test_memory_usage` as the others don't really branch for logic just relevant to indices or series. I put the series-specific logic of this test into two new ones.

I kept narrow_series and series_with_simple_index separated as I don't want to create another combined fixture with a very narrow use case.

Please have another look

File: pandas/tests/base/test_ops.py [hunk 1]
```diff
@@ -797,62 +798,88 @@ def test_fillna(self):
...
-            assert i[-1] == s.iloc[-1]
+    @pytest.mark.parametrize("invalid_value", [1, "True", [1, 2, 3], 5.0])
+    def test_validate_bool_args(self, invalid_value, series_with_simple_index):
+        series = series_with_simple_index
+        msg = "expected type bool"
+        with pytest.raises(ValueError, match=msg):
+            series.drop_duplicates(inplace=invalid_value)
 
-            assert i[-1] == i[9]
+    def test_getitem(self, indices):
```
Comment (jbrockmendel) [inline pandas/tests/base/test_ops.py]: this test looks to be in good shape but could use a more informative name

File: pandas/tests/base/test_ops.py [hunk 2]
```diff
@@ -797,62 +798,88 @@ def test_fillna(self):
...
-        for i in self.indexes:
-            s = pd.Series(i)
+        index = np.searchsorted(obj, max_obj, sorter=range(len(obj)))
+        assert 0 <= index <= len(obj)
 
-            assert i[0] == s.iloc[0]
-            assert i[5] == s.iloc[5]
-            assert i[-1] == s.iloc[-1]
+    @pytest.mark.parametrize("invalid_value", [1, "True", [1, 2, 3], 5.0])
+    def test_validate_bool_args(self, invalid_value, series_with_simple_index):
```
Comment (jbrockmendel) [inline pandas/tests/base/test_ops.py]: can the name reflect that it is about "drop_duplicates"?  also this should work for any Series, not just those with simple_index?

possibly belongs in tests.series.test_validate?

File: pandas/tests/base/test_ops.py [hunk 3]
```diff
@@ -797,62 +798,88 @@ def test_fillna(self):
...
+        assert total_usage == non_index_usage + index_usage
+
+    def test_searchsorted(self, index_or_series_obj):
+        # See gh-12238
+        obj = index_or_series_obj
 
-    def test_validate_bool_args(self):
-        invalid_values = [1, "True", [1, 2, 3], 5.0]
+        if isinstance(obj, pd.MultiIndex):
+            pytest.skip("np.searchsorted doesn't work on pd.MultiIndex")
```
Comment (jbrockmendel) [inline pandas/tests/base/test_ops.py]: is this something that should work but doesnt, or will never work?

File: pandas/tests/base/test_ops.py [hunk 4]
```diff
@@ -797,62 +798,88 @@ def test_fillna(self):
...
+        assert total_usage == non_index_usage + index_usage
+
+    def test_memory_usage_components_narrow_series(self, narrow_series):
+        series = narrow_series
+        total_usage = series.memory_usage(index=True)
+        non_index_usage = series.memory_usage(index=False)
+        index_usage = series.index.memory_usage()
+        assert total_usage == non_index_usage + index_usage
+
+    def test_searchsorted(self, index_or_series_obj):
```
Comment (jbrockmendel) [inline pandas/tests/base/test_ops.py]: name distinguish that this is about np.searchsorted, not obj.searchsorted

File: pandas/tests/base/test_ops.py (hunk 2 above)
Comment (SaturnFromTitan) [reply pandas/tests/base/test_ops.py]: Indeed, good catch! I removed this one and extended the parametrization in test_validate.py

File: pandas/tests/base/test_ops.py (hunk 3 above)
Comment (SaturnFromTitan) [reply pandas/tests/base/test_ops.py]: It's bugged, see #14833. I also added that reference as a comment

File: pandas/tests/base/test_ops.py (hunk 4 above)
Comment (SaturnFromTitan) [reply pandas/tests/base/test_ops.py]: turns out `np.searchsorted` actually calls `obj.searchsorted` under the hood... I added a comment to highlight this surprising fact

**Comment by jreback** [comment]
thanks @SaturnFromTitan

**Pull request merged**